### ISSUE TITLE
Remove the "doNotClean" message from all masterlists

### ISSUE THREAD

**Issue by pStyl3** [loot/loot#1407]
Plugins with ITMs (Identical to Master), Deleted References or Deleted Navmeshes are considered `dirty` and we advise users to clean those plugins with xEdit.

However, there are some plugins where mod authors are of the opinion, that those ITMs are `intentional` and as such, shouldn't be cleaned.

If we find such a plugin, that only has ITMs (which are claimed to be intentional) and no other problems (like Deleted References and Deleted Navmeshes), we have a couple of ways to handle the situation:

- Don't actively tell our users about the cleaning report
- Report the identified ITMs nonetheless
- Tell our users to not clean the plugin in question (`doNotClean`)

As of today, the masterlists have the following number of plugins with the `doNotClean` message attached:
Skyrim: 48
SSE: 4
Oblivion: 31
Morrowind: 0
FO3: 1
FO4: 0
FNV: 8

92 uses of `doNotClean`.

From a strictly technical point of view, the option to report ITMs nonetheless is the sentiment that I would likely share the most. It has been stated by multiple people, that ITMs aren't needed to achieve some change, so in other words, that "Intentional ITMs" do not really have valid use cases, or are the best way of approaching things.

From the point of view of LOOT, it isn't quite that easy though. As of today, we provide the ability to tell users to `doNotClean` a mod. In my understanding, this has been made practise so that mod authors do have a say in the matter. While we trust xEdit, there are people who may disagree, so this possibility opens up the situation for them.

That being said, I'm leaning more and more towards removing the `doNotClean` messages altogether. I'd rather give mod authors of plugins which only have ITMs and no other problems, where they claim those ITMs are intentional, the option that we don't actively tell users about those ITMs, than telling the users to not clean the plugins. That way we can keep a middle ground of some sorts, but don't further act on the narrative that cleaning "intentional" ITMs can lead to issues.

So in conclusion, I think we should remove the `doNotClean` messages. Plugins that only have ITMs and where those ITMs are claimed to be intentional could be left "untouched" by the masterlist, if the mod author so wishes, but don't actively tell users to not clean plugins. Thoughs?

**Comment by Infernio** [comment]
We should probably leave a comment in every place we drop the `doNotClean`, so that later contributors don't try adding the cleaning info in those places.

**Comment by MacSplody** [comment]
There is a way to make most ITMs into intentional changes by modifying the editorId of the record.
xEdit won't treat them as ITMs in that case and everyone viewing the file will know it's deliberate change.
I say most records, Landscape records for example don't have an editorID.

As an aside, another case where ITMs can spring up, is in files that are missing masters.
An example of that is one of RLO patches in the SSE masterlist.

So in most of these cases, it's an issue the author can rectify.

**Comment by pStyl3** [comment]
I think it's a good idea, that in case we go forward with removing the `doNotClean` messages from the masterlists, we leave a comment on the plugins that communicates this circumstance. So for example, instead of
```yaml
  - name: 'Example.esp'
    msg: [ *doNotClean ]
```
use
```yaml
  - name: 'Example.esp'
    # doNotClean
```
By reading this it should then be inherently clear to the masterlist maintainer, that this plugin
- likely has ITMs
- but no Deleted References and Deleted Navmeshes (having these invalidates an authors wish towards us to not communicate the cleaning report, in my opinion)
- the author thinks of the ITMs as `intentional`
- the author has explicitly expressed the wish that LOOT doesn't tell it's users about these ITMs
- and as such, we don't actively communicate these ITMs (but also do *not* tell users that they shouldn't clean them)

@Sharlikran also has pointed out on Discord, what Elminster, the main developer of xEdit, has once said regarding the topic of Intentional ITMs: [Link](https://tes5edit.github.io/docs/7-mod-cleaning-and-error-checking.html#TherearenointentionalITMs)
`If your mod contains intentional ITMs, then there is something wrong with it. If you have a specific conflict with a specific mod and you create a compatibility patch with ITMs and that other mod as master, then your ITMs will become identical to master, but conflict winner and don't get cleaned, ergo no intentional ITMs.`

This statement is one of the strongest reasons against the concept of `Intentional ITMs`, and as such, further strengthens my believe to remove the `doNotClean` messages in LOOT's masterlists.

All that being said, I've checked the masterlists, if any of them have the `doNotClean` message attached to official plugins or unofficial patches. I've found that the following plugins have the message attached to them:

__Fallout 3:__
Fallout3.esm
Anchorage.esm
ThePitt.esm
BrokenSteel.esm
PointLookout.esm
Zeta.esm

__Oblivion:__
Unofficial Oblivion Patch.esp

What was the reason for us to tell users to not clean the official Fallout 3 plugins? I remember doing exactly that (as always) and successfully playing through the entire game with no real issues a couple of years ago.

**Comment by Infernio** [comment]
TTW, most likely?

**Comment by MacSplody** [comment]
> Oblivion:
> Unofficial Oblivion Patch.esp

 I think it still had a few ITMs, is UOP safe to clean now? @arthmoor

> Fallout 3:
> Fallout3.esm
> Anchorage.esm
> ThePitt.esm
> BrokenSteel.esm
> PointLookout.esm
> Zeta.esm
> 
> What was the reason for us to tell users to not clean the official Fallout 3 plugins? 

If it's in the Fallout NV masterlist then yes it's for Tale of Two Wastelands.

**Comment by pStyl3** [comment]
> If it's in the Fallout NV masterlist then yes it's for Tale of Two Wastelands.
Oops, yes those entries are in the FNV masterlist, not the FO3 masterlist.

**Comment by Arthmoor** [comment]
There should be nothing to clean in the UOP. The only intended ITM was marked with an EDID change to prevent xEdit from removing it.

It was only ever an issue for SI users anyway.

**Comment by Arthmoor** [comment]
As far as not cleaning official master files, I'm of the opinion these days that it's probably not beneficial enough to do it now given the risk of an unknown bug in xEdit causing problems with the content in those files. I think that advice should apply to any game since there have been issues reported with every game xEdit supports except Oblivion. Yes, they wrote workarounds for the issues into the xEdit code, but that doesn't mean other similar issues won't crop up again later.

**Comment by Infernio** [comment]
Also, the workarounds are only in 4.0.3d and newer. The latest release on Nexus is still 4.0.3.

**Comment by MacSplody** [comment]
> 
> 
> There should be nothing to clean in the UOP. The only intended ITM was marked with an EDID change to prevent xEdit from removing it.
> 
> It was only ever an issue for SI users anyway.


```yaml
  - name: 'Unofficial Oblivion Patch.esp'
    dirty:
      - <<: *quickClean
        crc: 0x6EBDF804
        util: '[TES4Edit v4.0.3f](https://www.nexusmods.com/oblivion/mods/11536)'
        itm: 5
    clean:
      - crc: 0xF58C1B30
        util: 'TES4Edit v4.0.3f'
```

> [00:53] Done: Applying Filter, [Pass 1] Processed Records: 104300, [Pass 2] Processed Records: 97272, Remaining unfiltered nodes: 97272, Elapsed Time: 00:46
> [Undeleting and Disabling References done]  Processed Records: 97271, Undeleted Records: 0, Elapsed Time: 00:00
> Removing: [CELL:00085681] (in DreamWorld02 "Grotto of Courage" [WRLD:00085680])
> Removing: DreamWorld02 "Grotto of Courage" [WRLD:00085680]
> Removing: [CELL:00071F7E] (in KvatchEntrance "Kvatch" [WRLD:00071F7C])
> Removing: [CELL:0004A288] (in ICImperialPalaceMQ16 "Imperial City, Green Emperor Way" [WRLD:0004A250])
> Removing: [CELL:00045F96] (in MQ14OblivionWorld "Plane of Oblivion" [WRLD:00045F94])
> [Removing "Identical to Master" records done]  Processed Records: 97271, Removed Records: 5, Elapsed Time: 00:00

I left it as doNotClean as I wasn't sure on those.

**Comment by Arthmoor** [comment]
Where did you get that copy of the UOP? Whenever I check it I come back with a clean report.

`[00:24] Done: Applying Filter, [Pass 1] Processed Records: 104300, [Pass 2] Processed Records: 97272, Remaining unfiltered nodes: 97272, Elapsed Time: 00:24
[Removing "Identical to Master" records done]  Processed Records: 97271, Removed Records: 0, Elapsed Time: 00:00
[Undeleting and Disabling References done]  Processed Records: 97271, Undeleted Records: 0, Elapsed Time: 00:00`

**Comment by MacSplody** [comment]
> 
> 
> Where did you get that copy of the UOP? Whenever I check it I come back with a clean report.
> 
> `[00:24] Done: Applying Filter, [Pass 1] Processed Records: 104300, [Pass 2] Processed Records: 97272, Remaining unfiltered nodes: 97272, Elapsed Time: 00:24 [Removing "Identical to Master" records done] Processed Records: 97271, Removed Records: 0, Elapsed Time: 00:00 [Undeleting and Disabling References done] Processed Records: 97271, Undeleted Records: 0, Elapsed Time: 00:00`

https://www.nexusmods.com/oblivion/mods/5296?tab=files

I cleaned with default settings in xEdit and startup argument `-qac`.

**Comment by Arthmoor** [comment]
Ah, ok. I'm still very much a big fan of the traditional method. And before you dismiss that out of hand, yes, I know how to get the same results.

The problem here is that what QAC is calling an ITM doesn't mean the strict definition of an ITM. These cells are records with actual changes, but changes that are unnecessary. They're harmless CK noise. The kind of thing that would be very likely to come right back in the next time the mod is saved in the CK.

**Comment by MacSplody** [comment]
> Ah, ok. I'm still very much a big fan of the traditional method. And before you dismiss that out of hand, yes, I know how to get the same results.
> 
> The problem here is that what QAC is calling an ITM doesn't mean the strict definition of an ITM. These cells are records with actual changes, but changes that are unnecessary. They're harmless CK noise. The kind of thing that would be very likely to come right back in the next time the mod is saved in the CK.

Yeah I got the same result as you with manual and simple records on.
Not sure how having simple records turned allows removal of a grid value,
It looks the same as below in both cases.
Actually it's only QAC that will remove the value, manual with simple records turned off leaves it as well.
![image](https://user-images.githubusercontent.com/1944639/106371147-fa260680-6358-11eb-8221-1425178641ca.png)

**Comment by Arthmoor** [comment]
I always have it set up to turn simple records off, because that's how QAC assumes things should be done.

If this is supposed to be logically the same as removing superfluous location data from Skyrim REFR records, then the manual method should have been updated to handle this too to keep things consistent. Since both types of issues are things each CK will put back when the files are saved there.

**Comment by pStyl3** [comment]
Let's summarise the situation, as it is now.

__Official Plugins__
Currently we actively communicate xEdit cleaning reports, and don't advise to `doNotClean` them.

__Official FO3 plugins in FNV masterlist__
This is the one exception regarding official plugins. For TTW (apparently), we advise to `doNotClean` the official FO3 plugins within the NV masterlist. However, the official NV plugins do not get the same treatment, we actively (and unconditionally) communicate xEdit cleaning reports for them and don't advise to `doNotClean` them.

__Unofficial Patches__
The only one where we advise to `doNotClean` the Unofficial Patch is within the Oblivion masterlist, for `Unofficial Oblivion Patch.esp`.

___
When it comes to "normal" mod plugins, it's my belief that we should stop to actively tell users to `doNotClean` them. As laid out, the most we should do, is to not actively communicate cleaning reports for plugins, that only have ITMs and where those ITMs are seen by the mod author as `intentional`, if the mod author so wishes. This is a middle ground for us, so that mod authors do have a say in this, as long as their plugins do not have Deleted References or Deleted Navmeshes, but we should stop to further support the narrative that plugins shouldn't be cleaned because of `intentional ITMs`, and as such stop advising to `doNotClean`.

Regarding Official plugins, we don't advise to `doNotClean` them and communicate xEdit cleaning reports for them. Arthmoor and certainly others aswell have said, that they rather would prefer that the official plugins would not get cleaned, because there will always be a risk of an oversight in the xEdit cleaning procedure. This could be discussed in another issue if wanted, but within the narrative of this issue, we don't really need to do anything, because we don't advise to `doNotClean` the official plugins anyway. And *if* we should decide at some point in the future, that cleaning the official plugins represents a greater risk than not cleaning them, then the cleaning reports should simply be deleted from the masterlists. Even in this case I would say that we should not advise to `doNotClean` them.

Regarding the one exception, the official FO3 plugins within the NV masterlist. Here we advise to `doNotClean` them, as it is with `TaleOfTwoWastelands.esm` aswell. While it seems to be mandatory for TTW that both games are in their original state and thus the official plugins shouldn't be cleaned (unless I'm mistaken), I don't think that this warrants a special treatment within our masterlists. It is the responsibility of TTW that their users start with both games being prestine and vanilla. And as we currenlly don't actively advise to `doNotClean` the official NV plugins anyway, I think that in this case aswell, that we should remove the `doNotClean` messages.

Last but not least regarding the `Unofficial Oblivion Patch.esp`, the only unofficial patch where we advise to `doNotClean` the plugin, it seems that it is clean or if xEdit is set up to find them, it's only `CK noise` that get's found, as Arthmoor puts it. As such, I see no problem to remove the `doNotClean` message here.

___
In conclusion, I think there's no major issue with removing the `doNotClean` messages.

**Comment by MacSplody** [comment]
> Regarding the one exception, the official FO3 plugins within the NV masterlist. Here we advise to doNotClean them, as it is with TaleOfTwoWastelands.esm aswell.

It's said enough times in the TTW notes and multiple guides not to clean its files.

> While it seems to be mandatory for TTW that both games are in their original state and thus the official plugins shouldn't be cleaned (unless I'm mistaken),

The TTW installer will abort if it detects cleaned DLC, so that's kind of already being handled.
It could be done If we could add a condition to hide dirty entries, I don't think we can atm.

**Comment by Arthmoor** [comment]
I have no opinion on the TTW issue since I don't use it and have done basically nothing as far as modding FO3 or NV in general.

The UOP thing is kind of an odd thing since QAC mode is the only way those 5 records are going to get removed, and leaving them alone poses no harm since it's just the CK saving data specifically to indicate what its absence would also indicate. I could always issue an update to the UOP to put that one to bed but if we have cause to update it again a year or two down the road, nobody is going to remember to housekeep those records again.

**Comment by pStyl3** [comment]
I will then go ahead and remove the `doNotClean` messages.

**Comment by pStyl3** [comment]
With the merging of the above linked pull requests, the issue has been resolved.

**Comment by sibir-ine** [comment]
@pStyl3 By the way, there's a filter for hiding doNotClean messages that has been made redundant by it's removal.

![image](https://user-images.githubusercontent.com/69141501/107853294-e1423a00-6dc9-11eb-9fef-bf0815bec118.png)

**Comment by pStyl3** [comment]
You are absolutely correct. This filter should probably removed, now that we no longer tell users to `doNotClean` plugins.

**Comment by Ortham** [comment]
Closing this as I've removed the filter in e456714d375d78910d042983e35159bd9d5535f2.
